holmes/plugins/toolsets/robusta/robusta.py

```diff
@@ -59,6 +65,53 @@ def get_parameterized_one_liner(self, params: Dict) -> str:
         return "Fetch metadata and history"
 
 
+class FetchConfigurationChanges(Tool):
+    _dal: Optional[SupabaseDal]
+
+    def __init__(self, dal: Optional[SupabaseDal]):
+        super().__init__(
+            name="fetch_configuration_changes",
+            description="Fetch configuration changes in a given time range. By default, fetch all cluster changes. Can be filtered on a given namespace or a specific workload",
+            parameters={
+                START_TIME: ToolParameter(
+                    description="The starting time boundary for the search period. String in RFC3339 format.",
+                    type="string",
+                    required=True,
+                ),
+                END_TIME: ToolParameter(
+                    description="The starting time boundary for the search period. String in RFC3339 format.",
+                    type="string",
+                    required=True,
+                ),
+            },
+        )
+        self._dal = dal
+
+    def _fetch_change_history(self, params: Dict) -> Optional[List[Dict]]:
+        if self._dal and self._dal.enabled:
+            return self._dal.get_configuration_changes(
+                start_datetime=params["start_datetime"],
+                end_datetime=params["end_datetime"],
+            )
+        return None
+
+    def _invoke(self, params: Dict) -> str:
+        try:
+            changes = self._fetch_change_history(params)
+            if changes:
+                return yaml.dump(changes)
+            else:
+                return f"Could not find changes for {params}"
+        except Exception as e:
+            logging.error(e)
+            msg = f"There was an internal error while fetching changes for {params}. {str(e)}"
+            logging.error(msg)
```

Comment:
Please change to exception (Or add stack trace) + duplicate error logging (Also 2 line above.

Reply:
fixed